Sources/PortalSwift/Provider/PortalProvider.swift

```diff
@@ -214,82 +177,255 @@ public class PortalProvider {
 
   /// Makes a request.
   /// - Parameters:
-  ///   - payload: A normal payload whose params are of type [Any].
-  ///   - completion: Resolves with a Result.
-  /// - Returns: Void
+  ///   - chainId: A CAIP-2 Blockchain ID associated with the request.
+  ///   - withMethod: A member of the PortalRequestMethod enum
+  ///   - andParams: An array of parameters for the request (either RPC parameters or a transaction if signing)
+  /// - Returns: PortalProviderResult
   public func request(
-    payload: ETHRequestPayload,
-    completion: @escaping (Result<RequestCompletionResult>) -> Void,
+    _ chainId: String,
+    withMethod: PortalRequestMethod,
+    andParams: [AnyEncodable]? = [],
     connect: PortalConnect? = nil
-  ) {
-    let isSignerMethod = signerMethods.contains(payload.method)
+  ) async throws -> PortalProviderResult {
+    let blockchain = try PortalBlockchain(fromChainId: chainId)
+    guard blockchain.isMethodSupported(withMethod) else {
+      throw PortalProviderError.unsupportedRequestMethod(withMethod.rawValue)
+    }
+
     let id = UUID().uuidString
 
-    // Handle changing chains
-    if payload.method == ETHRequestMethods.WalletSwitchEthereumChain.rawValue {
-      let param = payload.params[0] as! [String: String]
+    // This switch is here to handle methods that should be
+    // resolved by the provider directly before passing the
+    // request on to RPC or the signer.
+    //
+    // The default behavior is to use the `PortalBlockchain`
+    // instance to determine if the method should be signed
+    // or not.
+    switch withMethod {
+    case .eth_accounts, .eth_requestAccounts:
+      let address = try await keychain.getAddress(chainId)
+      return PortalProviderResult(id: id, result: [address])
+    case .wallet_switchEthereumChain:
+      return PortalProviderResult(id: id, result: "null")
+    default:
+      if blockchain.shouldMethodBeSigned(withMethod) {
+        let payload = PortalProviderRequestWithId(id: id, method: withMethod, params: andParams)
+        return try await self.handleSignRequest(chainId, withPayload: payload, forId: id, onBlockchain: blockchain, connect: connect)
+      } else {
+        return try await self.handleRpcRequest(chainId, withMethod: withMethod, andParams: andParams, forId: id)
+      }
+    }
+  }
 
-      if let chainId = UInt8(param["chainId"]!.replacingOccurrences(of: "0x", with: ""), radix: 16) {
-        do {
-          _ = try self.setChainId(value: Int(chainId), connect: connect)
+  /// Makes a request.
+  /// - Parameters:
+  ///   - chainId: A CAIP-2 Blockchain ID associated with the request.
+  ///   - withMethod: The string literal of your RPC method
+  ///   - andParams: An array of parameters for the request (either RPC parameters or a transaction if signing)
+  /// - Returns: PortalProviderResult
```

Comment:
A helper `request` overload that accepts a `String` as the method and attempts to map to our enum.